commitizen-tools/commitizen
# Lotram/customizable allowed prefixes

## Description

Related to https://github.com/commitizen-tools/commitizen/issues/672
The allowed message prefix list can now be configured. 
In particular, it can be set to an empty list, if we do not want to allow commits such as fixups.

I also fixed a typo in `.pre-commit-config.yaml` and a typing error in `commitizen/cli.py`
## Checklist


## Expected behavior

The allowed prefix list can be configured, either using a config file, or when using the cli directly. 

## Steps to Test This Pull Request
1. `poetry shell` 
1. Run `python commitizen/cli.py check --message 'fixup!'  --allowed-prefixes 'Merge' 'Revert' 'Custom Prefix'`. It should return an error
2. Run `python commitizen/cli.py check --message "Custom Prefix random text"  --allowed-prefixes 'Merge' 'Revert' 'Custom Prefix'`. It should pass


## Additional context
